src/app/InteractionModelEngine.h

```diff
@@ -43,17 +40,21 @@
 #include <app/Command.h>
 #include <app/CommandHandler.h>
 #include <app/CommandSender.h>
+#include <app/InteractionModelDelegate.h>
+#include <app/ReadClient.h>
+#include <app/ReadHandler.h>
 
-#define CHIP_MAX_NUM_COMMAND_HANDLER_OBJECTS 1
-#define CHIP_MAX_NUM_COMMAND_SENDER_OBJECTS 1
+#define CHIP_MAX_NUM_COMMAND_HANDLER 1
+#define CHIP_MAX_NUM_COMMAND_SENDER 1
+#define CHIP_MAX_NUM_READ_CLIENT 1
```

Comment:
It appears this caps the number of concurrent reads to 1.  Setting to something like #CHIP_CONFIG_MAX_DEVICE_ADMINS would seem a lot safer.

I suppose this can work if all reads are completely synchronous.  But I'm concerned that this will show an artificially small memory footprint for this feature, when in production we may need much more.

If we are finding that we need a lot of some class of object, have we considered a memory pool as we have for pbufs?  Obviously this is more complex to implement for objects, but it's still possible.  It seems like the interaction model may create a need for many concurrent interactions, but yet it may be difficult to predict how many of each type would be needed.

Reply:
Read Client could be device A that read attributes from device B, and may be not administrator that provision the device.
I would file one ticket to investigate if the memory pool is good for IM's client and handler.

Reply:
#5440